src/seedu/addressbook/commands/FindCommand.java

```diff
@@ -50,8 +50,12 @@ public CommandResult execute() {
         final List<ReadOnlyPerson> matchedPersons = new ArrayList<>();
         for (ReadOnlyPerson person : addressBook.getAllPersons()) {
             final Set<String> wordsInName = new HashSet<>(person.getName().getWordsInName());
-            if (!Collections.disjoint(wordsInName, keywords)) {
-                matchedPersons.add(person);
+            for(String wordInName: wordsInName) {
+                for(String keyword: keywords) {
+                    if(wordInName.equalsIgnoreCase(keyword) || wordInName.contains(keyword)) {
```

Comment:
here you didn't only igonre the case, but also allow to search keyword as substring of the word in name, but here you doesn't support case insensitive anymore. Thus, should only implement case insensitive first, then the partial search in another new branch.